### ISSUE TITLE
[Bug] Image animation not working in GTK

### ISSUE THREAD

**Issue by ysangkok** [xamarin/Xamarin.Forms#8732]
### Description
It seems like image animation is not working in GTK. Similar code worked on the UWP backend.

### Steps to Reproduce
https://github.com/ysangkok/XamarinGTKAnimationIssue

[src.zip](https://github.com/xamarin/Xamarin.Forms/files/3935547/src.zip)

### Expected Behavior
Image should be animated.

### Actual Behavior
Image is not animated.

### Basic Information

- Version with issue: 4.4.0.991210-pre2
- Last known good version: don't know
- Platform Target Frameworks: Ubuntu 18.04 with gtk-sharp 2.12.45-0xamarin17+ubuntu1804
- Nuget Packages: See linked repository

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 1.045 ETH (150.03 USD @ $143.57/ETH)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $76,903.23 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 4 weeks, 1 day from now.
Please review their action plans below:


**1) [izoch](https://gitcoin.co/izoch) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?mutate_worker_action=approve&worker=izoch) | [reject worker](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?mutate_worker_action=reject&worker=izoch))_.**

Job application test. I will probably start work on this this upcoming week-end probably. Work has been a bit hectic for the past few weeks.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888).

**Comment by gitcoinbot** [comment]
@izoch Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=1) | [3 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=3) | [5 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=5) | [10 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=10) | [100 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=100)

**Comment by Izoch** [comment]
Commenting back

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@izoch due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=1) | [3 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=3) | [5 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=5) | [10 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=10) | [100 days](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?snooze=100)

**Comment by mul53** [comment]
@samhouts does this require work? Would like to work on it

**Comment by knocte** [comment]
@mul53 what do you mean?

**Comment by mul53** [comment]
@knocte working on the bug

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 5 days, 16 hours ago.
Please review their action plans below:


**1) [bobface](https://gitcoin.co/bobface) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?mutate_worker_action=approve&worker=bobface) | [reject worker](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888?mutate_worker_action=reject&worker=bobface))_.**

I can fix the bug with the animation.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 1.045 ETH (272.64 USD @ $260.9/ETH) has been submitted by__:

 1. [@bobface](https://gitcoin.co/bobface)



@knocte please take a look at the submitted work:
* [PR](https://github.com/xamarin/Xamarin.Forms/pull/9649) by @Bobface

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/3888).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $82,169.00 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by ysangkok** [comment]
I tested this, with the sample project from the issue. It works, but it consumes 100% CPU and is not smooth. I am not sure whether this is a problem with this PR or with the sample project (which I made).

**Comment by Bobface** [comment]
Hmm, would it be possible to link the image you have used?

**Comment by Bobface** [comment]
@ysangkok Also, the large the image the more expensive the calculation for scale & rotate. Did you test if it works smoothly with a small image, say 50x50?

**Comment by ysangkok** [comment]
@Bobface I have a HiDPI monitor, so 50x50 is really small. But you are right that smaller sizes have less CPU load. For example, a 250x188 image runs with only 50% CPU utilization. I am using the ImageMagick logo image. To generate it in its full size (with width 500px), use `convert logo: logo.png`.

**Comment by knocte** [comment]
@Bobface maybe have a look at these answers: https://stackoverflow.com/questions/6888965/any-way-to-speed-up-reduce-cpu-usage-when-drawing-with-cairo?rq=1 :)

**Comment by Bobface** [comment]
@ysangkok Okay, so the size of the image is definitely the problem.
@knocte Thanks, but it seems like I already do everything as described in that thread.

**Comment by knocte** [comment]
Even the FPS reduction?

**Comment by Bobface** [comment]
That should be achievable by reducing the frequency rate of the animation in @ysangkok 's repro (`dummyFrequencyRate` in `LoadingPage.xaml.fs`). However he links to another issue which says that that parameter is also not implemented in Xamarin.Forms and has no effect. Is that still correct @ysangkok ?

**Comment by ysangkok** [comment]
@Bobface I have tested your PR 9856 with this one cherry-picked on top. Everything works as expected. Thank you.

**Comment by knocte** [comment]
@ysangkok you mean CPU doesn't go to 100% anymore?

**Comment by ysangkok** [comment]
@knocte CPU consumption is not too high with reasonable animation update intervals, no.

**Comment by Solexplorer** [comment]
Hi @ysangkok,

Is the fix provided by @Bobface working?

**Comment by ysangkok** [comment]
@Solexplorer yes

**Comment by Solexplorer** [comment]
@knocte Does this bounty also need review from maintainers?

**Comment by knocte** [comment]
Yes but this should be merged only after PR 9856 is merged.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This issue now has a funding of 1.045 ETH (249.23 USD @ $238.5/ETH)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/4541).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/4541).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>
* $307,869.71 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 month from now.
Please review their action plans below:


**1) [bobface](https://gitcoin.co/bobface) has started work.**

.................................................................

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/4541).

**Comment by gitcoinbot** [comment]
⚡️ A tip worth 1.04000  ETH (248.04 USD @ $238.5/ETH) has been granted to @bobface for this issue from @knocte. ⚡️ 

Nice work @bobface! Your tip has automatically been deposited in the ETH address we have on file.

 * $307873.51 in Funded OSS Work Available at: https://gitcoin.co/explorer
 * Incentivize contributions to your repo: <a href='https://gitcoin.co/tip'>Send a Tip</a> or <a href='https://gitcoin.co/funding/new'>Fund a PR</a>
 * No Email? Get help on the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>

**Comment by knocte** [comment]
Fixed by https://github.com/xamarin/Xamarin.Forms/pull/9649 which was merged , @ysangkok you can close

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This Bounty has been completed.__

Additional Tips for this Bounty:
* knocte tipped 1.0400 ETH worth 248.04 USD to bobface.

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/xamarin/Xamarin.Forms/8732/4541)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://chat.gitcoin.co/'>Gitcoin Chat</a>
* $307,873.51 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
